mtp_api/apps/security/models.py

```diff
@@ -376,6 +383,44 @@ def __str__(self):
         return f'Check {self.status} for {self.credit}'
 
 
+class CheckAutoAcceptRule(TimeStampedModel):
+    debit_card_sender_details = models.ForeignKey(
+        DebitCardSenderDetails, on_delete=models.CASCADE, related_name='check_auto_accept_rules'
+    )
+    prisoner_profile = models.ForeignKey(
+        PrisonerProfile, on_delete=models.CASCADE, related_name='check_auto_accept_rules'
+    )
+
+    objects = CheckAutoAcceptRuleManager()
+
+    def get_latest_state(self):
+        return self.states.order_by('-created').first()
+
+    def is_active(self):
+        return self.get_latest_state().active
+
+    class Meta:
+        ordering = ('-created',)
+        unique_together = (
+            ('debit_card_sender_details', 'prisoner_profile',),
+        )
+
+
+class CheckAutoAcceptRuleState(TimeStampedModel):
```

Comment:
Do you reckon is it also worth adding a default ordering by `created DESC` for the states? It seems like the natural orders, e.g. latest has priority over older states.

Reply:
Thinking about it, i'm not sure that `created DESC` makes any sense outside of the context of `get_latest_state`. Adding `created` ordering on `CheckAutoAcceptRuleState` and changing `CheckAutoAcceptRule` from `created DESC` to `created` sounds good though.